src/core/control/Control.cpp

```diff
@@ -1045,6 +1048,34 @@ void Control::clearSelectionEndText() {
     }
 }
 
+void Control::selectAllOnPage() {
+    auto pageNr = getCurrentPageNo();
+    if (pageNr == npos) {
+        return;
+    }
+
+    XojPageView* view = win->getXournal()->getViewFor(pageNr);
+    if (view == nullptr) {
+        return;
+    }
+
+    this->doc->lock();
```

Comment:
One last thing: I think the document should be locked already for this step.
```suggestion
    this->doc->lock();
    XojPageView* view = win->getXournal()->getViewFor(pageNr);
    if (view == nullptr) {
        this->doc->unlock();
        return;
    }

```